modules/service/src/main/scala/lucuma/odb/service/CalibrationsService.scala

```diff
@@ -31,13 +49,15 @@ trait CalibrationsService[F[_]] {
   def recalculateCalibrations(
     pid: Program.Id
   )(using Transaction[F]): F[Unit]
-
 }
 
 object CalibrationsService {
-  def instantiate[F[_]: Monad](using Services[F]): CalibrationsService[F] =
+  type GmosNConfigs = (GmosNorthGrating, GmosNorthFpu, Wavelength, Option[GmosXBinning], Option[GmosYBinning])
+  type GmosSConfigs = (GmosSouthGrating, GmosSouthFpu, Wavelength, Option[GmosXBinning], Option[GmosYBinning])
```

Comment:
It feels like the binning may be problematic.  It's never really `None`, but rather defaults when not explicitly defined. `None` can actually mean any of the binning options.

Reply:
From my tests i see this always defined, I think when you set the mode some binning will be set

However on the table is nullable thus the type needs to be `Option[A]`  I think

Reply:
If the PI doesn't go out of their way to set the binning, it will be calculated from a variety of parameters in code.  It's a pretty convoluted calculation.  Maybe we can figure out how to add the result to a view, I'm not sure.

If you do a query to get all the unique configurations, it seems like we need the actual binning that will be used.

Reply:
As we discussed we'll tackle this next on a separate PR